tests/phpunit/Unit/PackageType/PackageTest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+declare ( strict_types = 1 );
+
+namespace SatisPress\Test\Unit\PackageType;
+
+use Psr\Log\NullLogger;
+use SatisPress\Archiver;
+use SatisPress\Exception\PackageNotInstalled;
+use SatisPress\Package;
+use SatisPress\PackageType\BasePackage;
+use SatisPress\PackageType\PackageBuilder;
+use SatisPress\ReleaseManager;
+use SatisPress\Storage\Local as LocalStorage;
+use SatisPress\Test\Unit\TestCase;
+
+class PackageTest extends TestCase {
+	public function setUp(): void {
+		parent::setUp();
+
+		$archiver = new Archiver( new NullLogger() );
+		$storage  = new LocalStorage( SATISPRESS_TESTS_DIR . '/Fixture/wp-content/uploads/satispress/packages' );
+		$manager  = new ReleaseManager( $storage, $archiver );
+		$package  = new BasePackage();
```

Comment:
This seems like a lot of moving parts to add in, when the class being unit tested is PackageBuilder.

Could BasePackage and Manager be mocked? How much of PackageBuilder is being tested against the implementation of BasePackage, vs inserting a known value into a Mock and checking against that?

Reply:
I originally mocked the manager for this particular test case, but the extra dependencies were needed for some of the other cases that tested releases, so I just made the setup the same across all of them. It could be easily swapped for a mock in this particular case, though.

The PackageBuilder and BasePackage tests are kinda dependent on each other right now. I'll go ahead and split those out into separate test cases.